### DISCUSSION TITLE
Authenticate using third-party service JWT instead of NextAuth generated JWT

### DISCUSSION THREAD

**Discussion by Cuttsy27** [nextauthjs/next-auth#4177] [Help]
Hi,

I'm working on a project where I am essentially using the NextJS api page as a middleware/proxy to another backend we are building for the project. The other backend generates its own JWT which the NextJS client will use to authenticate. Is it possible to use the credentials provider login, get the JWT from our other backend service, and use it for the NextAuth session, instead of NextAuth providing a JWT for us?

If not, what could be another solution? I can store our backend JWT inside the NextAuth-provided JWT, but then we have multiple sources of truth to determine whether the user is "authenticated" or not, if the two JWTs drop out of sync.

**Comment by blyefd** [comment]
This is a good question. I also would like to know how to do this.

**Comment by Historia2** [comment]
No answer for this?

**Comment by MuhammadZunnoorainRafi** [comment]
https://cloudcoders.xyz/blog/nextauth-credentials-provider-with-external-api-and-login-page/
